packages/widgets/src/lib/components/InferenceWidget/widgets/ConversationalWidget/ConversationalWidget.svelte

```diff
@@ -51,26 +43,64 @@
 	let outputJson: string;
 	let text = "";
 
-	async function getOutput({
-		withModelLoading = false,
-		isOnLoadCall = false,
-		exampleOutput = undefined,
-	}: InferenceRunOpts = {}) {
+	let compiledTemplate: Template;
+
+	async function getOutput({ withModelLoading = false, isOnLoadCall = false }: InferenceRunOpts = {}) {
 		const trimmedText = text.trim();
 
 		if (!trimmedText) {
 			return;
 		}
 
-		if (shouldUpdateUrl && !conversation.past_user_inputs.length) {
+		// Error checking
+		const config = model.config;
+		if (config === undefined) {
+			outputJson = "";
+			output = [];
+			error = "Model config not found";
+			return;
+		}
+
+		const tokenizerConfig = config.tokenizer as TokenizerConfig | undefined;
+		if (tokenizerConfig === undefined) {
+			outputJson = "";
+			output = [];
+			error = "Tokenizer config not found";
+			return;
+		}
+
+		if (!compiledTemplate) {
+			const chatTemplate = tokenizerConfig.chat_template;
+			if (chatTemplate === undefined) {
+				outputJson = "";
+				output = [];
+				error = "No chat template found in tokenizer config";
+				return;
+			}
+			compiledTemplate = new Template(chatTemplate);
```

Comment:
Not sure if possible but should we use these conditions to determine whether to show the `text-generation` or the `conversational` widget? Repos such as 

* https://huggingface.co/cognitivecomputations/dolphin-2.5-mixtral-8x7b
* https://huggingface.co/microsoft/DialoGPT-medium
* https://huggingface.co/JosephusCheung/Guanaco

are `conversational` without templates. On the other hand, this seems like a great way to nudge them to add a template to the `tokenizer_config.json`, so this could be quite nice! cc @Rocketknight1

Reply:
I would tend to show the `conversational` widget with an explicit error to advocate for chat templates. It's not just for the widget but also because we plan to support chat-based inference in Python's client (and most likely JS as well?) which will rely on `config.tokenizer.chat_template`.

Also as mentioned in https://github.com/huggingface/huggingface.js/pull/457#issuecomment-1940905549, I think it would make sense to inject a default chat_template from moon-landing when applicable (e.g. [when we know the model type](https://github.com/huggingface/huggingface.js/blob/59e31011f0b1ea8d8cdf66dbe6cc2a341101ef29/packages/widgets/src/lib/components/InferenceWidget/widgets/ConversationalWidget/ChatTemplate.ts)) to reduce the number of conversational models without a chat template.

Reply:
> I would tend to show the conversational widget with an explicit error to advocate for chat templates

I agree with this

> I think it would make sense to inject a default chat_template from moon-landing when applicable 

I'm not sure about this one though - I think @julien-c also has an opinion

Reply:
> I would tend to show the conversational widget with an explicit error to advocate for chat templates

yes i agree too

> I think it would make sense to inject a default chat_template from moon-landing when applicable

not sure either atm. Roughly how many models do you think it concerns on the hub?

Reply:
Let's open PRs on the `microsoft/DialoGPT-***` repos though. @LysandreJik @Rocketknight1 do you know the chat template for those? Couldn't find it in the model card.

Reply:
yes, let's open PRs then! (we could even programmatically open a few PRs if we wanted to)

Reply:
Done! To keep the diffs short, I only append a `chat_template` key to the tokenizer_config.json. This also means it shouldn't break backwards compatibility. [For example](https://huggingface.co/nisten/BigCodeLlama-92b/discussions/2/files):
![image](https://github.com/huggingface/huggingface.js/assets/26504141/dc702cd3-9ab8-4238-ad5a-d4e5d35026e9)


<details>

<summary>See links</summary>

https://huggingface.co/0xDEADBEA7/DialoGPT-small-rick/discussions/2#65cf7a9f00d099dc4cb7093f
https://huggingface.co/AI-Sweden-Models/gpt-sw3-1.3b-instruct/discussions/7#65cf7a3a70354febcb50d421
https://huggingface.co/AI-Sweden-Models/gpt-sw3-20b-instruct-4bit-gptq/discussions/5#65cf7a7f8050d6c98389ed81
https://huggingface.co/AI-Sweden-Models/gpt-sw3-20b-instruct/discussions/12#65cf7a4a6130ef7be061eb8f
https://huggingface.co/AI-Sweden-Models/gpt-sw3-356m-instruct/discussions/6#65cf7a4217e2b305e0ae6987
https://huggingface.co/AI-Sweden-Models/gpt-sw3-6.7b-v2-instruct/discussions/7#65cf7a48bda9379ede3176d2
https://huggingface.co/AVeryRealHuman/DialoGPT-small-TonyStark/discussions/2#65cf78312d592af8eabba2b4
https://huggingface.co/BlackSamorez/rudialogpt3_medium_based_on_gpt2_2ch/discussions/3#65cf7a9aad23a674049a902a
https://huggingface.co/Fredithefish/Guanaco-13B-Uncensored/discussions/3#65cf7a4d70354febcb50d8b7
https://huggingface.co/Fredithefish/Guanaco-3B-Uncensored-v2/discussions/4#65cf7a47c1aba98e9a1723d5
https://huggingface.co/Fredithefish/Guanaco-3B-Uncensored/discussions/4#65cf7a4fb104eb084a41952d
https://huggingface.co/Fredithefish/Guanaco-7B-Uncensored/discussions/3#65cf7a5798b4a470bfcc4b58
https://huggingface.co/IlyaGusev/saiga_mistral_7b_merged/discussions/3#65cf7a738050d6c98389ea27
https://huggingface.co/JamesStratford/Pidrow-bot-DialoGPT-Large-Feb2023/discussions/3#65cf7a7c0954f06e47dbb2de
https://huggingface.co/L-R/LLmRA-3B-v0.1/discussions/2#65cf7a43053a863f53069e07
https://huggingface.co/L-R/LLmRa-1.3B/discussions/7#65cf7a6150818a335a1f507a
https://huggingface.co/L-R/LLmRa-1.3B_V2/discussions/3#65cf7a5f092ee72727d7195a
https://huggingface.co/L-R/LLmRa-2.7B/discussions/4#65cf7a402fd803be40393736
https://huggingface.co/LTC-AI-Labs/L2-7b-Base-WVG-Uncensored/discussions/3#65cf7a3d6130ef7be061e929
https://huggingface.co/LTC-AI-Labs/L2-7b-Hermes-Synthia/discussions/2#65cf7a3bad23a674049a78ad
https://huggingface.co/Lazycuber/Janemalion-6B/discussions/3#65cf7a5b244b552be0a7f213
https://huggingface.co/Locutusque/gpt2-conversational-or-qa/discussions/7#65cf7a51244b552be0a7f01e
https://huggingface.co/Neko-Institute-of-Science/pygmalion-7b/discussions/6#65cf7a53ad23a674049a7de5
https://huggingface.co/OpenLLM-France/Claire-7B-0.1/discussions/2#65cf7a94b4ed8768781d9322
https://huggingface.co/OpenLLM-France/Claire-Mistral-7B-0.1/discussions/1#65cf7aa3891646a27b34beea
https://huggingface.co/Pi3141/DialoGPT-medium-elon-2/discussions/2#65cf782417e2b305e0adf792
https://huggingface.co/Pi3141/DialoGPT-medium-elon-2/discussions/3#65cf7a298050d6c98389d76b
https://huggingface.co/PygmalionAI/pygmalion-1.3b/discussions/8#65cf7a4598b4a470bfcc4769
https://huggingface.co/PygmalionAI/pygmalion-2.7b/discussions/5#65cf7a3695e8d86e2f418cf7
https://huggingface.co/PygmalionAI/pygmalion-350m/discussions/6#65cf7a3fad23a674049a7969
https://huggingface.co/PygmalionAI/pygmalion-6b/discussions/47#65cf7a27f8ab7fc669d1ca33
https://huggingface.co/RUCAIBox/mvp/discussions/3#65cf7a33818f059346db0b5f
https://huggingface.co/ScyKindness/Hatsune_Miku/discussions/2#65cf7aa559ef10734ff4bd27
https://huggingface.co/Severus27/BeingWell_llama2_7b/discussions/2#65cf7a7188d13d8128b919f5
https://huggingface.co/SotirisLegkas/final_socratic_dialoGPT/discussions/2#65cf7a9b563be22cb964f29b
https://huggingface.co/TehVenom/Pygmalion-13b-Merged/discussions/3#65cf7a5688d13d8128b91597
https://huggingface.co/TheBloke/Guanaco-3B-Uncensored-v2-GPTQ/discussions/4#65cf7a6b298f19fe951bf97d
https://huggingface.co/TheBloke/goliath-120b-AWQ/discussions/1#65cf7aa1563be22cb964f3d3
https://huggingface.co/TheBloke/goliath-120b-GPTQ/discussions/3#65cf7a781e5f505ebbfd06f7
https://huggingface.co/VMware/open-llama-0.7T-7B-open-instruct-v1.1/discussions/3#65cf7a63bf849f3b79fadab1
https://huggingface.co/aari1995/germeo-7b-laser/discussions/2#65cf7a2b030f1c957015e172
https://huggingface.co/ademfatnassi/bonjourGPT-small/discussions/4#65cf7a9895e8d86e2f41a481
https://huggingface.co/af1tang/personaGPT/discussions/7#65cf7a26af96705574fa1dd8
https://huggingface.co/alpindale/goliath-120b/discussions/19#65cf7a59818f059346db2529
https://huggingface.co/ayjays132/PhillnetLarge/discussions/2#65cf7aa617e2b305e0ae7be1
https://huggingface.co/ayjays132/phillnet/discussions/1#65cf7a8c092ee72727d736d3
https://huggingface.co/bardsai/jaskier-7b-dpo/discussions/2#65cf7a675de2b19b9fe4f479
https://huggingface.co/bit-dny/MindLLM-1b3-chat-zh-v2.0/discussions/2#65cf7a5d59ef10734ff4aaf7
https://huggingface.co/cedpsam/chatbot_fr/discussions/2#65cf7a9d6130ef7be061fc57
https://huggingface.co/core42/jais-13b-chat/discussions/23#65cf7a24563be22cb964da2e
https://huggingface.co/core42/jais-30b-chat-v1/discussions/7#65cf7a2e2d592af8eabc143b
https://huggingface.co/damerajee/Oot-v2_lll/discussions/3#65cf7a6f053a863f5306bad1
https://huggingface.co/deepparag/Aeona/discussions/7#65cf7a6dbda9379ede317f20
https://huggingface.co/derek-thomas/jais-13b-chat-hf/discussions/3#65cf7a30b104eb084a418f57
https://huggingface.co/garrachonr/LlamaDos/discussions/2#65cf7a929a8e7192d2a452bf
https://huggingface.co/gorkemgoknar/gpt2chatbotenglish/discussions/4#65cf7a90bda9379ede318833
https://huggingface.co/h2oai/h2ogpt-gm-oasst1-en-2048-falcon-7b-v2/discussions/9#65cf7a2c0954f06e47dba214
https://huggingface.co/hakurei/mommygpt-3B/discussions/2#65cf7a65c5eb016dcbeaba44
https://huggingface.co/ingen51/DialoGPT-medium-GPT4/discussions/5#65cf7a7a1e4ed74045f7a943
https://huggingface.co/llSourcell/medllama2_7b/discussions/6#65cf7a7d70354febcb50e2f5
https://huggingface.co/llmixer/BigWeave-v12-90b/discussions/1#65cf7a8ac5eb016dcbeac21b
https://huggingface.co/llmixer/BigWeave-v15-103b/discussions/1#65cf7a95294aa5a9686cbd30
https://huggingface.co/llmixer/BigWeave-v6-90b/discussions/1#65cf7a8ea838056fd6e869cc
https://huggingface.co/microsoft/DialoGPT-large/discussions/16#65cf7a21298f19fe951bc199
https://huggingface.co/microsoft/DialoGPT-medium/discussions/15#65cf7a1ec1aba98e9a171a06
https://huggingface.co/microsoft/DialoGPT-small/discussions/8#65cf7a2370354febcb50cf43
https://huggingface.co/mlinmg/SG-Raccoon-Yi-55B-200k-2.0/discussions/2#65cf7933092ee72727d6def3
https://huggingface.co/mlinmg/SG-Raccoon-Yi-55B-200k-2.0/discussions/3#65cf7a748e145354382822cf
https://huggingface.co/mywateriswet/ShuanBot/discussions/2#65cf783500d099dc4cb65084
https://huggingface.co/mywateriswet/ShuanBot/discussions/3#65cf7a3800d099dc4cb6c73d
https://huggingface.co/nisten/BigCodeLlama-169b/discussions/2#65cf7a869e6805a7b1642d14
https://huggingface.co/nisten/BigCodeLlama-92b/discussions/2#65cf7a8117e2b305e0ae7501
https://huggingface.co/ostorc/Conversational_Spanish_GPT/discussions/2#65cf7a842fd803be40394840
https://huggingface.co/satvikag/chatbot/discussions/5#65cf7a325de2b19b9fe4e734
https://huggingface.co/timpal0l/Mistral-7B-v0.1-flashback-v2/discussions/2#65cf7a6900d099dc4cb6e49b
https://huggingface.co/tinkoff-ai/ruDialoGPT-small/discussions/3#65cf7a765de2b19b9fe4f8ec
https://huggingface.co/xHexyy/test3/discussions/3#65cf7a83c5eb016dcbeac090

</details>

Reply:
(i've merged the ones on `microsoft/DialoGPT-{small, medium, large}` as we used to kinda co-maintain those models

cc @LysandreJik @younesbelkada for viz^

Reply:
Nice!